ddtrace/contrib/botocore/patch.py

```diff
@@ -52,23 +53,87 @@ def inject_trace_data_to_message_attributes(trace_data, entry):
         log.debug("skipping trace injection, max number (10) of MessageAttributes exceeded")
 
 
-def inject_trace_to_sqs_batch_message(args, span):
+def inject_trace_to_sqs_or_sns_batch_message(args, span):
     trace_data = {}
     HTTPPropagator.inject(span.context, trace_data)
     params = args[1]
 
-    for entry in params["Entries"]:
+    entries = params.get("Entries", params.get("PublishBatchRequestEntries", []))
+    for entry in entries:
         inject_trace_data_to_message_attributes(trace_data, entry)
 
 
-def inject_trace_to_sqs_message(args, span):
+def inject_trace_to_sqs_or_sns_message(args, span):
     trace_data = {}
     HTTPPropagator.inject(span.context, trace_data)
     params = args[1]
 
     inject_trace_data_to_message_attributes(trace_data, params)
 
 
+def inject_trace_to_event_bridge_detail(args, span):
+    params = args[1]
+    if "Entries" in params:
+        for entry in params["Entries"]:
+            if "Detail" in entry:
+                try:
+                    detail = json.loads(entry["Detail"])
+                    dd_context = detail.get("_datadog", {})
+                    HTTPPropagator.inject(span.context, dd_context)
+                    detail["_datadog"] = dd_context
+                    entry["Detail"] = json.dumps(detail)
+                except Exception:
+                    log.warning("Unable to parse Detail and inject span context")
+            else:
+                detail = {}
+                HTTPPropagator.inject(span.context, detail)
+                entry["Detail"] = json.dumps(detail)
+    else:
+        log.debug("Unable to inject context. The Event Bridge event had no Entries.")
+
+
+def get_kinesis_data_object(data, try_b64=True):
+
+    # check if data size will exceed max with headers
```

Comment:
What is this max we don't want to exceed? Is it the max payload (json) size that we can return?

Reply:
Yes, the max size for a kinesis record supported by AWS is 1MB.